components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
It's not just about end-user scripts, but our own scripts. If you search for `patient.document`, it's used in several places in our codebase.